cloud/db/project-memberships.ts

```diff
@@ -0,0 +1,845 @@
+    DrizzleORM
+  > {
+    return Effect.gen(this, function* () {
+      const client = yield* DrizzleORM;
+
+      yield* this.authorize({
+        userId,
+        action: "read",
+        organizationId,
+        projectId,
+        memberId: userId,
+      });
+
+      const results: PublicProjectMembership[] = yield* client
+        .select(publicFields)
+        .from(projectMemberships)
+        .where(eq(projectMemberships.projectId, projectId))
+        .pipe(
+          Effect.mapError(
+            (e) =>
+              new DatabaseError({
+                message: "Failed to find all project memberships",
+                cause: e,
+              }),
+          ),
+        );
+
+      return results;
+    });
+  }
+
+  /**
+   * Retrieves a specific membership by member ID.
+   *
+   * Requires project membership (any role) or implicit access from org OWNER/ADMIN.
+   *
+   * @param args.userId - The authenticated user
+   * @param args.organizationId - The organization containing the project
+   * @param args.projectId - The project
+   * @param args.memberId - The user ID of the member to retrieve
+   * @returns The membership with role and creation date
+   * @throws NotFoundError - If the membership doesn't exist
+   * @throws PermissionDeniedError - If the user lacks read permission
+   * @throws DatabaseError - If the database query fails
+   */
+  findById({
+    userId,
+    organizationId,
+    projectId,
+    memberId,
+  }: {
+    userId: string;
+    organizationId: string;
+    projectId: string;
+    memberId: string;
+  }): Effect.Effect<
+    PublicProjectMembership,
+    NotFoundError | PermissionDeniedError | DatabaseError,
+    DrizzleORM
+  > {
+    return Effect.gen(this, function* () {
+      const client = yield* DrizzleORM;
+
+      yield* this.authorize({
+        userId,
+        action: "read",
+        organizationId,
+        projectId,
+        memberId: userId,
+      });
+
+      const [membership] = yield* client
+        .select(publicFields)
+        .from(projectMemberships)
+        .where(
+          and(
+            eq(projectMemberships.projectId, projectId),
+            eq(projectMemberships.memberId, memberId),
+          ),
+        )
+        .limit(1)
+        .pipe(
+          Effect.mapError(
+            (e) =>
+              new DatabaseError({
+                message: "Failed to find project membership",
+                cause: e,
+              }),
+          ),
+        );
+
+      if (!membership) {
+        return yield* Effect.fail(
+          new NotFoundError({
+            message: `Membership for member ${memberId} not found`,
+            resource: this.getResourceName(),
+          }),
+        );
+      }
+
+      return membership;
+    });
+  }
+
+  /**
+   * Updates a membership's role.
+   *
+   * Requires ADMIN role (implicit org OWNER/ADMIN or explicit project ADMIN).
+   * Cannot be used to change your own role (no self-modification).
```

Comment:
I thought we agreed that it's okay for org owner/admin to explicitly add themselves to projects